### ISSUE TITLE
Support and options for compressing (gzip, brotli etc.)

### ISSUE THREAD

**Issue by ammnt** [AdguardTeam/AdGuardHome#6026]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to request a feature or enhancement and not ask a question

### The problem

Hello,

please, consider to add more compressing settings for DoH and control panel such as:
- support of compressing like gzip and brotli;
- control of it and level of compressing too;
- control of size of buffers, HTTP headers for this and etc..

Thank you.
Cheers😜

### Proposed solution

Described before.

**Comment by ainar-g** [comment]
I'm a bit wary about turning AGH into a full-blown Nginx replacement.  We'll see what kind of reactions this request is going to get, but anyone who needs customization of these kinds of HTTP parameters is better off just using a reverse proxy.

**Comment by fernvenue** [comment]
> I'm a bit wary about turning AGH into a full-blown Nginx replacement.

I agree, I don't think AdGuardHome should do these kind of thing by itself.
